Vine: Added an obspy example

arctangerines opened the pull request:
## Proposed Changes

Added a taskvine example that uses ObsPy to process seismic data using some MiniSEED files available in the examples repository provided by ObsPy.
Also a small typo fix.

## Merge Checklist

The following items must be completed before PRs can be merged.
Check these off to verify you have completed all steps.

- [ ] `make test`       Run local tests prior to pushing.
- [ ] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [ ] `make lint`       Run lint on source code prior to pushing.
- [ ] Manual Update:     Update the manual to reflect user-visible changes.
- [ ] Type Labels:       Select a github label for the type: bugfix, enhancement, etc.
- [ ] Product Labels:    Select a github label for the product: TaskVine, Makeflow, etc.
- [ ] PR RTM:            Mark your PR as ready to merge.

btovar commented on taskvine/src/examples/vine_example_obspy.py:
```diff
@@ -0,0 +1,96 @@
+import ndcctools.taskvine as vine
+
+seed2png = """
```
I suggest converting this into a function `seed2png(..)` and using `PythonTask` instead of Task. This has the advantage that python can detect errors in the `seed2png` code before sending the tasks.

arctangerines replied:
Ok sent the changes.
Some interesting things happened:
1. Initially it wasn't working, miniSEED file was being parsed wrong for some reason
2. Not only that but matplotlib was complaining it couldn't find a display, despite the fact we are using the "Agg" backend which requires no display

After I moved the matplotlib and co imports inside the function it started working, afterwards I moved the obspy import, just for coherence.

I am going to make a semi educated guess and say that this was happening because when we called PythonTask, it didn't have complete visibility of the imports, or something along those lines(?).

arctangerines commented:
Ok sent the changes.
Some interesting things happened:
1. Initially it wasn't working, miniSEED file was being parsed wrong for some reason
2. Not only that but matplotlib was complaining it couldn't find a display, despite the fact we are using the "Agg" backend which requires no display

After I moved the matplotlib and co imports inside the function it started working, afterwards I moved the obspy import, just for coherence.

I am going to make a semi educated guess and say that this was happening because when we called PythonTask, it didn't have complete visibility of the imports, or something along those lines(?).

arctangerines commented on taskvine/src/examples/vine_example_obspy.py:
```diff
@@ -0,0 +1,81 @@
+
+
+def seed2png(filename):
+    from obspy import read
```
Hmmm, so I decided to use poncho since we already have it available in the ndcctools package and it seems like it downloads all the environments correctly butttttt it seems to be hitting a wall with poncho:
```
Traceback (most recent call last):
   File "/tmp/worker-1000-461063/task.1/w_bf7b0e45-c08a-4c57-8ee4-9f563e177cf3", line 3, in <module>
    import cloudpickle
ModuleNotFoundError: No module named 'cloudpickle'
```

Will investigate further, to see what's going on...

dthain replied:
`cloud pickle` should be part of the packages in `environment.yml`, yes?

arctangerines replied:
Hi, sorry for the delay but even adding cloudpickle gives us this error:
```python
Traceback (most recent call last):
  File "/tmp/worker-1000-104647/task.5/w_834503d9-685b-45b9-b9cb-dd617a4a4814", line 7, in <module>
    exec_function = cloudpickle.load(f)
```
Tried to grep the repo and find the files that have a line like this and it's [task.py](https://github.com/cooperative-computing-lab/cctools/blob/master/taskvine/src/bindings/python3/ndcctools/taskvine/task.py#L1034) and [futures.py](https://github.com/cooperative-computing-lab/cctools/blob/master/taskvine/src/bindings/python3/ndcctools/taskvine/task.py#L1034), not sure exactly which one is the culprit but I'm going to take a very obvious guess and say it's `task.py`. 
Wonder if I can inspect what the parameters the error talks about are...

btovar replied:
Are the python and cloudpickle versions in the env the same as in the poncho spec? What is your poncho spec?

arctangerines replied:
That was it, the python version was 3.10 in the poncho environment, and the environment was 3.13. Thanks, will send PR once I'm back on my computer and I test out some things.

arctangerines commented:
Ok so here are the changes, also added a shebang for python andddddd fixed an error with the import in the `vine_example_poncho_in_python.py` file
